Dockerfile

```diff
@@ -23,34 +23,39 @@
 #
 
-FROM fnndsc/ubuntu-python3:latest
-MAINTAINER fnndsc "dev@babymri.org"
-
-# Pass a UID on build command line (see above) to set internal UID
-ARG UID=1001
-ENV UID=$UID DEBIAN_FRONTEND=noninteractive APPROOT="/home/localuser/pfcon"
-
-RUN apt-get update                                                                              \
-  && apt-get install -y libssl-dev libcurl4-openssl-dev bsdmainutils net-tools inetutils-ping   \
-  && apt-get install -y locales                                                                 \
-  && export LANGUAGE=en_US.UTF-8                                                                \
-  && export LANG=en_US.UTF-8                                                                    \
-  && export LC_ALL=en_US.UTF-8                                                                  \
-  && locale-gen en_US.UTF-8                                                                     \
-  && dpkg-reconfigure locales  && pip install --upgrade pip                                     \
-  && useradd -u $UID -ms /bin/bash localuser
-
-# Copy source code
-COPY --chown=localuser ./bin ${APPROOT}/bin
-COPY --chown=localuser ./pfcon ${APPROOT}/pfcon
-COPY --chown=localuser ./setup.cfg ./setup.py README.rst  ${APPROOT}/
-
-RUN pip3 install ${APPROOT}  \
-  && rm -fr ${APPROOT}
-
-# Start as user localuser
-#USER localuser
-
-WORKDIR "/home/localuser"
-ENTRYPOINT ["pfcon"]
-EXPOSE 5005
+FROM python:3.8.6-buster AS build
+  MAINTAINER fnndsc "dev@babymri.org"
+
+  # Pass a UID on build command line (see above) to set internal UID
+  ARG UID=1001
+  ENV UID=$UID DEBIAN_FRONTEND=noninteractive APPROOT="/home/localuser/pfcon"
+
+  RUN apt-get update                                                                              \
+    && apt-get install -y --no-install-recommends libssl-dev libcurl4-openssl-dev bsdmainutils    \
+       net-tools inetutils-ping locales                                                           \
+    && export LANGUAGE=en_US.UTF-8                                                                \
+    && export LANG=en_US.UTF-8                                                                    \
+    && export LC_ALL=en_US.UTF-8                                                                  \
+    && locale-gen en_US.UTF-8                                                                     \
+    && dpkg-reconfigure locales  && pip install --upgrade pip pytest                              \
+    && useradd -u $UID -ms /bin/bash localuser
+
+  # Copy source code
+  COPY --chown=localuser ./setup.cfg ./setup.py README.rst ./requirements.txt ${APPROOT}/
+  RUN pip3 install -r ${APPROOT}/requirements.txt
+
+  COPY --chown=localuser ./bin ${APPROOT}/bin
+  COPY --chown=localuser ./pfcon ${APPROOT}/pfcon
+
+  RUN pip3 install --no-dependencies ${APPROOT}  
+
+FROM build as tests
+
+  RUN pytest ${APPROOT}/pfcon/tests/*.py
+  RUN rm -fr ${APPROOT}
```

Comment:
`RUN rm -fr ${APPROOT}` is an effort to minimize the published image size, so it should be part of the `runtime` stage. Though it might be futile because it's adding a layer.

Reply:
Good catch. Yes, I should have moved it to the `runtime` stage. But as you noted, we're not gaining much in terms of image size; that step takes a noticeable amount of time during building, however. So I'm going to remove it.

To optimize the image size, we may want to adjust the CI and Dockerfile as indicated [here](https://pythonspeed.com/articles/faster-multi-stage-builds/). I'll leave this to a future task as that matter isn't as pressing, though :)